config/txgh.yml

```diff
@@ -1,15 +0,0 @@
-txgh:
-    github:
-        repos:
-            matthewjackowski/txgh-test-resources:
-                api_username: <%= ENV['GITHUB_USERNAME']  %>
-                api_token: <%= ENV['GITHUB_TOKEN'] %>
-                push_source_to: <%= ENV['GITHUB_PUSH_SOURCE_TO'] %>
-                branch: <%= ENV['GITHUB_BRANCH'] %>
-    transifex:
-        projects:
-            txgh-test-1:
-                tx_config: <%= ENV['TX_CONFIG_PATH'] %>
-                api_username: <%= ENV['TX_USERNAME'] %>
-                api_password: <%= ENV['TX_PASSWORD'] %>
-                push_translations_to: <%= ENV['TX_PUSH_TRANSLATIONS_TO'] %>
```

Comment:
I think I would feel more comfortable separating the application and configuration concerns entirely. Right now the config is checked in to the txgh repo, but the config is very organization-specific. Configuration seems like an organization concern whereas txgh itself is more of a library maintained by multiple organizations.

Since we're currently deploying instances of txgh with Docker at Lumos, all our config is given via environment variables, including the contents of the tx.config file (in fact, tx.config is part of each git repo and downloaded on deploy). To support this, I've slightly changed the way txgh looks for config (changes are in my "docker" branch). Instead of looking inside itself, txgh looks for a config directory inside the current working directory. I thought that was more flexible since it gives the user the opportunity to run txgh from a separate directory with the right config (think multi-tenancy). Eventually I would like to be able to provide another environment variable with the path to tx.config.

The other config file, txgh.yml, is entirely made up of environment variables. I'm thinking we can eventually move it into the code and delete the actual file.

Publishing txgh as a gem will help in your Heroku case. You should be able to create a separate, organization-specific repo (eg. transifex-txgh) that contains *example* config, a Gemfile with the txgh gem in it, and a config.ru file that tells rack how to run the application.

Thoughts?